src/main/scala/com/nec/spark/agile/SparkExpressionToCExpression.scala

```diff
@@ -95,6 +97,9 @@ object SparkExpressionToCExpression {
         ar.withName(s"input_${idx}->data[left_out[i]]")
       case idx if (rightIds.contains(ar.exprId)) =>
         ar.withName(s"input_${idx}->data[right_out[i]]")
+      case _ => sys.error(s"SparkExpressionToCExpression.referenceReplacer: " +
```

Comment:
has this happened? Just wondering.

Reply:
Not sure. But seems like a good check to add.